python/tk_multi_loader/model_latestpublish.py

```diff
@@ -79,107 +79,129 @@ def load_data(self, item, child_folders, show_sub_items, additional_sg_filters):
+                    # but is switching into more of a paradigm of an inverse
+                    # database. Indicate the difference by not showing any folders
+                    child_folders = []
             else:
-                # standard mode - show folders and items for the currently selected item
+                # standard mode - show folders and items for the currently selected items
                 # for leaf nodes and for tree nodes which are connected to an entity,
                 # show matches.
 
-                # Extract the Shotgun data and field value from the node item.
-                (sg_data, field_value) = model_item_data.get_item_data(item)
-
-                if sg_data:
-                    # leaf node!
-                    # show the items associated. Handle tasks
-                    # via the task field instead of the entity field
-                    if sg_data.get("type") == "Task":
-                        sg_filters = [["task", "is", {"type": sg_data["type"], "id": sg_data["id"]}]]
-                    elif sg_data.get("type") == "Version":
-                        sg_filters = [["version", "is", {"type": "Version", "id": sg_data["id"]}]]
-                    else:
-                        sg_filters = [["entity", "is", {"type": sg_data["type"], "id": sg_data["id"]} ]]
-
-                else:
-                    # intermediate node.
+                # because we might have multiple entities selected, we need to create a list to hold all our filters
+                sg_filters = []
 
-                    if isinstance(field_value, dict) and "name" in field_value and "type" in field_value:
-                        # this is an intermediate node like a sequence or an asset which
-                        # can have publishes of its own associated
-                        sg_filters = [["entity", "is", field_value ]]
-
-                    else:
-                        # this is an intermediate node like status or asset type which does not
-                        # have any publishes of its own, because the value (e.g. the status or the asset type)
-                        # is nothing that you could link up a publish to.
+                # loop through the selected items
+                for item in items:
+                    if item is None:
+                        # nothing selected in the treeview
+                        # passing none to _load_data indicates that no query should be executed
                         sg_filters = None
-
+                    else:
+                        # Extract the Shotgun data and field value from the node item.
+                        (sg_data, field_value) = model_item_data.get_item_data(item)
+
+                        if sg_data:
+                            # leaf node!
+                            # show the items associated. Handle tasks
+                            # via the task field instead of the entity field
```

Comment:
can you add a period to the end of the old comment here and format the lines a bit for consistency.